### DISCUSSION TITLE
Syncing notes and journals not possible with ownCloud

### DISCUSSION THREAD

**Discussion by SigHunter** [TechbeeAT/jtxBoard#48] [Q&A] [answered]
hi,
I'm using jtx board 2.01.01-rc12.ose from fdroid together with davx5 from fdroid.

webdav server is an owncloud 10.11

in Collections it shows 
Journals: n. V. Notes: n. V. Tasks: 0

how can I sync Journals and Notes?
when creating J+N it is not possible to select davx synced target

edit: davx shows only calendar and tasks icon. but latest owncloud should support VJOURNAL, correct?

Thanks and Regards

S

**Comment by patrickunterwegs** [comment, marked as answer]
Hey @SigHunter , 
basically yes, I have just tried it again with my ownCloud 10 instance and it just worked. But as I'm only using it for testing purposes, I am using SQLite and not a dedicated database on a server. 

Collections have components enabled or disabled. In your case it seems likely that you're trying to use a collection, that only supports VTODO (and maybe VEVENT), but not VJOURNAL (which would enable journals and notes). 
Please try to make a new collection out of DAVx5 on your server and make sure that the Journals/Notes checkbox is activated. Check out the following post on how to do that: https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong

There was another user with a technical problem with ownCloud who came to the conclusion that there is a problem in the database of ownCloud: He manually had to alter the table "oc_calendar", the column "components" needed to be enlarged to take more characters. 
So if the problem persists and you're technically aware enough, you could also check this. 

Cheers, 
Patrick

**Reply by SigHunter** [reply]
thanks for your response!
interesting, I can see in davx debug log:
SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'components' at row 1

this is a relatively new owncloud instance (not older than a year) with mariadb (10.10) as db backend.
probably needs something like "alter table oc_calendars modify components varchar(some number);" is there a reasonable size?
will this break anything? currently its varchar(20)

creating a new calendar did not help btw (via oc web and davx), or did you mean a completely new sync account?

further down the rabbit hole: 
so "components" basically only contains the string VEVENT,VTODO (12 chars) and in this case needs to be VEVENT,VTODO,VJOURNAL with 21 chars, limit is 20, I'll increment it to 22 and see what happens ;)

**Reply by patrickunterwegs** [reply]
No, no, it was correct like that to just create a new calendar, creating a new user wouldn't change anything.

Ok, so the SQL error very much seems like the technical issue from the other user...
I need to have a closer look at that and also create a bug report on Owncloud

**Reply by patrickunterwegs** [reply]
Thanks for the update, it seems like it worked with the update, right? :-)
Do you still have the update statement, would you mind pasting it here for further reference?
I would also create an issue report on owncloud now

**Reply by patrickunterwegs** [reply]
Hey @SigHunter !
So I have opened the issue here: 
https://github.com/owncloud/web/issues/8109
Maybe you can also watch it if they have any questions ;-)
And I'd be grateful for the Update-SQL!

**Reply by SigHunter** [reply]
great, however I have not tried it yet, I'm a bit worried what this might break in owncloud and I rely quite a bit on my oc instance.
What was the response from the other guy who had this problem? any repercussions?
Is VJOURNAL better implemented in nextcloud?

**Reply by patrickunterwegs** [reply]
alright, no worries, but could you have a look at the issue I created for owncloud? Have I described it correctly as you experienced it? (I just have a bunch of work and I'd like to save the time to set up everything to reproduce it on my own  if possible, so I very much appreciate your support 🙂 )
I didn't follow up with the other guy on that specific problem, but he mentioned that he did the change (extended the column character length to 25) and it worked for him.

About Nextcloud: Basically I'm closer to Nextcloud, also my friends from DAVx5 are using Nextcloud and we have been at their conference this year. It works smoothly on Nextcloud, so there's no setup issue like you describe for owncloud. But apart from this, I wouldn't say that Nextcloud has a better integration, there's also no web-frontend for journals and notes (yet)...

**Reply by SigHunter** [reply]
small detail: it's oc_calendars
the rest is correct. 
I hope they don't complain about mariadb 10.10, I think it's not supported yet (but the db structure was created with some much older version some time last year, so that did not change anything)

thanks for your help, appreciate it very much and I'm looking forward to using jtx board

**Reply by patrickunterwegs** [reply]
Thanks, I corrected the table name in the ticket 👍 
And thank you too for your support on this issue 😉

**Comment by patrickunterwegs** [comment]
> edit: davx shows only calendar and tasks icon. but latest owncloud should support VJOURNAL, correct?

sorry, I didn't pay enough attention to your edit, so anyway, it is very likely that you just have to create a new collection with Journals/Notes support enabled 😉

**Comment by SigHunter** [comment]
I realized I don't actually need a webdav sync collection that can do everything. (VTODO, VJOURNAL, VEVENT) 
I actually prefer to not touch my owncloud calendar collection and I added a separate collection that has only VJOURNAL enabled for syncing notes! that should keep the chars <=20 easily
altering the columnsize helped too btw. for anyone who wants to try:
adapt to your needs, I have it in docker:
`docker compose down`
`cp -a mysql/ mysql.bak`
`docker compose up -d`
`docker exec -it owncloud_mariadb bash`
`mariadb -p`
`enter the dbroot pw`
`use owncloud;`
`show columns from oc_calendars;`
`alter table oc_calendars modify components varchar(255);`
`show columns from oc_calendars;`
`exit`
but I set it back to 20 to keep owncloud defaults, whatever reason they had for varchar(20)

**Reply by patrickunterwegs** [reply]
Hey @SigHunter , thanks for the update, yes, indeed that should work without problems then. You should also be able to use VJOURNAL and VTODO in one collection, this would allow you to use subtasks within Journals and Notes 😉 
In https://github.com/owncloud/core/issues/40537 somebody replied that they had reduced the field length of that DB-field to 20, obviously it's not very clear why. but as it was much longer before, I would say it wouldn't be any problem to keep it longer. But let's see what they say there in the owncloud ticket!
